quran/fields.py

```diff
@@ -0,0 +1,44 @@
+from django.core import exceptions
+from django.db import models
+
+
+class SegmentsField(models.TextField):
+    description = "A custom field to save list of timecodes in DB as a string"
+
+    def to_python(self, segments):
+        if isinstance(segments, list) or segments is None or segments == '':
+            return segments
+        try:
+            return [
+                tuple(int(timecode) for timecode in timecodes.split(':'))
+                for timecodes in segments.split(',')]
+        except (TypeError, ValueError):
+            raise exceptions.ValidationError(
+                message='Invalid string of segments',
+                code='invalid',
+            )
+
+    def from_db_value(self, value, expression, connection):
+        return self.to_python(value)
+
+    def get_prep_value(self, segments_list):
+        if segments_list is None or segments_list == '':
```

Comment:
Usually, people try converting everything `to_python()` first and then make conversions with no checks.